### PR TITLE
Enable bare except checking

### PR DESCRIPTION
**Pull request by abadger**
##### SUMMARY
```
except:
```
if generally a bad thing to do in Python.  It catches unexpected things like sys.exit() and KeyboardInterrupt.  Catching Exception ```except Exception:``` is usually considered bad style but at least it doesn't have the sys.exit and keyboardInterrupt problem so switch all occurrences of bare except catching to ```except Exception:```.  Also enable the pylint test for this so that we don't regress.

##### ISSUE TYPE

- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
Various

##### ANSIBLE VERSION

```
2.8
```

##### ADDITIONAL INFORMATION
At the moment, each individual site has not been reviewed to see if there is a reason to catch sys.exit and KeyBoardInterrupt.  I would like to get feedback from the other devs about areas which do not need review (I believe that modules, for instance, should never be catching those.  And, in fact, if they are, then there's very likely a bug in the code since AnsibleModule.exit_json and AnsibleModule.fail_json call sys.exit.  If there's an bare exception handler around code which calls fail_json fail_json won't trigger correctly)  Once we know which areas need review, I can review those pieces to make sure we don't need to explicitly catch some other exceptions in the port.

This should go in at the beginning of the 2.8 development cycle.  So within the next few weeks.

/cc @mattclay @nitzmahone @sivel

### CODE REVIEW

**Review by webknjaz — APPROVED** [review]
I like this change.

**Comment by abadger** [comment]
bot_skip

**Comment by abadger** [comment]
Risk reward analysis:

Controller side:

* Risk that we want to catch KeyboardInterrupt (for instance when processing user input) and the change would no longer do that.  (Explicitly catch KeyboardInterrupt instead of/in addition to catching Exception to fix that).
* Risk that we want to catch sys.exit (probably only if a badly written library that we depend on calls it) and (Explicitly catch SystemExit instead of/in addition to Exception to fix that).
* Reward: we won't be catching KeyboardInterrupt unintentionally.  Catching KeyboardInterrupt might be a user frustration if they're hitting Ctrl-C and nothing is happening.  However, we might have some of that frustration anyways due to our multi-worker architecture.

Module side:
* Risk that we want to catch sys.exit (probably only if badly written library that we depend on calls it).
* Reward: modules won't inadvertantly catch SystemExit from calling module.fail_json() further down in the call stack, which would end up doing further processing after a fail_json() call.

(KeyboardInterrupt shouldn't be either a pro or con module side as there shouldn't be a way to generate that module-side).

**Comment by mattclay** [comment]
@abadger How do you want to move this forward? It might be easier to enable the rule, add the necessary per-file skip entries and then submit separate PRs to fix the issues. Splitting up the fixes into multiple PRs would allow them to be grouped as needed to facilitate reviews by the appropriate people and keep slower reviews from holding up fixes for ones we're able to expedite.

**Comment by abadger** [comment]
I'd like to figure out the subset of this that can be fixed without input from other people because it's deemed safe or because it is code under the core teams responsibility anyway.  Then fixing those and implementing the ship for only those files that remain.  Leaving it up to other people to make changes won't make quick progress so I'm loathe to just leave that. 

Module side we probably should fix everything that's using bare except because it's likely that as we revamp basic.py, bare except will end up catching sys.exit/SystemExit in now instances... Fail_json won't be limited to places that have access to a module so it will become even more of a trip hazard than it currently is

File: test/runner/lib/thread.py
Comment (mattclay) [inline test/runner/lib/thread.py]: Keep the bare `except` here.

We want to catch all exceptions, even those that don't derive from `BaseException` under Python 2.x.

Of course there shouldn't be any of those, but just in case.

File: contrib/inventory/zabbix.py
Comment (mattclay) [inline contrib/inventory/zabbix.py]: Catch `Exception` and `SystemExit` here instead of `BaseException`.

Same for other occurrences in the Zabbix module

**Review by mattclay — CHANGES_REQUESTED** [review]
Looks good, other than a few minor changes and needing a rebase.

File: test/runner/lib/thread.py
Comment (abadger) [reply test/runner/lib/thread.py]: Good point!  I forgot that Python2 still has old-style-class exceptions that aren't rooted at BaseException.

File: contrib/inventory/zabbix.py
Comment (abadger) [reply contrib/inventory/zabbix.py]: Roger wilco.

**Comment by abadger** [comment]
Okay, pushed an update.  I'm trying to get this updated, tested by shippable, and merged so that it isn't conflicting with a bunch of weekday merges.

**Comment by abadger** [comment]
Checks are just unstable rather than failed and the errors look unrelated.  Merging to devel

**Pull request merged**
